### PR TITLE
to address issue #17, better explained settings, auto-copy settings f…

### PR DESCRIPTION
**Pull request by mg979**
…ile if not present, additional variables reset on close_window

### CODE REVIEW

File: Main.sublime-menu [hunk 1]
```diff
@@ -13,19 +13,13 @@
...
-                                "caption": "Settings – Default"
+                                "caption": "Edit settings"
                             },
-                            {
-                                "command": "open_file",
-                                "args": {
-                                    "file": "${packages}/User/code_map.sublime-settings"
-                                },
-                                "caption": "Settings – User"
-                            }
```
Comment (mg979) [inline Main.sublime-menu:28]: Unified settings command opens both base settings and user settings in new window

File: code_map.py [hunk 2]
```diff
@@ -81,6 +87,11 @@ def plugin_loaded():
         # package was installed manually
         plugin_dir = path.dirname(__file__)
 
+        # make a copy of the settings file in User folder, if not present
+        if not path.isfile(settings_file):
+            base_file = path.join(plugin_dir, 'code_map.sublime-settings')
+            shutil.copyfile(base_file, settings_file)
+
```
Comment (mg979) [inline code_map.py]: make copy of the settings file in the user folder if not already done

File: code_map.py [hunk 3]
```diff
@@ -136,8 +147,8 @@ def get_group(view):
         if w:
             return w.get_view_index(view)[0]
 
-    # I wish C# style checking was possible with Python: 
-    # return view?.window()?.get_view_index(view)[0]          
+    # I wish C# style checking was possible with Python:
+    # return view?.window()?.get_view_index(view)[0]
```
Comment (mg979) [inline code_map.py:142]: Please set:

`"trim_trailing_white_space_on_save": true,`

in `Preferences -> Settings` or at least in `Preferences -> Settings - Syntax specific`

PLEASE.. every time the diff looks like this. Python doesn't want trailing white spaces.

File: code_map.py [hunk 4]
```diff
@@ -935,7 +946,8 @@ def on_window_command(self, window, command_name, args):
                  "prompt_select_workspace",
                  "open_recent_project_or_workspace",
                  "close_workspace",
-                 "project_manager"]
+                 "project_manager",
+                 "close_window"]
```
Comment (mg979) [inline code_map.py]: added "close_window" to the list

File: code_map.py [hunk 5]
```diff
@@ -87,11 +82,6 @@ def plugin_loaded():
         # package was installed manually
         plugin_dir = path.dirname(__file__)
 
-        # make a copy of the settings file in User folder, if not present
-        if not path.isfile(settings_file):
-            base_file = path.join(plugin_dir, 'code_map.sublime-settings')
-            shutil.copyfile(base_file, settings_file)
-
         for syntax in default_mappers:
```
Comment (mg979) [inline code_map.py]: removed because if settings change across versions it can cause problems.

**Comment by oleg-shilo** [comment]
All done, thank you.

**Comment by mg979** [comment]
I proposed those changes so hopefully it's less likely that it is inactivated and not reactivated. It was a bit irritating with that deactivation method. I didn't have problems for now. Thanks

**Pull request merged**